assets/src/amp-story-editor-blocks.js

```diff
@@ -0,0 +1,237 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { compose, createHigherOrderComponent } from '@wordpress/compose';
+import domReady from '@wordpress/dom-ready';
+import { setDefaultBlockName } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { withAttributes, withParentBlock, withBlockName, withHasSelectedInnerblock, withAmpStorySettings } from './components';
+import { ALLOWED_BLOCKS, BLOCK_TAG_MAPPING } from './helpers';
+
+domReady( () => {
+	addFilter( 'blocks.registerBlockType', 'ampStoryEditorBlocks/addAttributes', addAMPAttributes );
+	addFilter( 'blocks.registerBlockType', 'ampStoryEditorBlocks/setBlockParent', setBlockParent );
+	addFilter( 'editor.BlockEdit', 'ampStoryEditorBlocks/filterEdit', withAmpStorySettings );
+	addFilter( 'editor.BlockListBlock', 'ampStoryEditorBlocks/addWrapperProps', addWrapperProps );
+	addFilter( 'blocks.getSaveContent.extraProps', 'ampStoryEditorBlocks/addExtraAttributes', addAMPExtraProps );
+
+	/**
+	 * Change the default block type to amp/amp-story-page.
+	 *
+	 * This way no paragraph blocks gets inserted when pressing enter in the <PostTitle> component.
+	 *
+	 * Ideally we could override the post title so that pressing enter results in a new block inserted
+	 * inside the first page block.
+	 *
+	 * Todo: set default block name to text block when already inside
+	 */
+	setDefaultBlockName( 'amp/amp-story-page' );
```

Comment:
I just discovered that this breaks slash command insertion.

Reply:
It breaks slash commands only when `'amp/amp-story-page'`. If I change to `'core/paragraph'` then it seems all good.